### PR TITLE
SignMessage updated and added CL & Eclair backends

### PR DESCRIPTION
**Pull request by pseudozach**
# Description

Relates to issue: https://github.com/ZeusLN/zeus/pull/500#issuecomment-887765899

This PR resolves lnurl-auth specification compliance issue when Zeus uses LND v2 signmessage as mentioned in above comment. It also adds support for other backends.

This pull request is categorized as a:

- [ ] New feature
- [ ] Bug fix
- [X] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [X] I’ve run `npm run tsc` and make sure my code didn’t produce errors 
- [X] I’ve run `npm run prettier` and formatted my code correctly

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [X] No, I’m a fool
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [X] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [X] LND
- [X] c-lightning-REST
- [ ] spark
- [X] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `npm install` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: backends/CLightningREST.ts [hunk 1]
```diff
@@ -115,6 +115,10 @@ export default class CLightningREST extends LND {
         });
     getRoutes = () => this.getRequest('N/A');
     getUTXOs = () => this.getRequest('/v1/listFunds');
+    signMessage = (message: any) =>
```
Comment (kaloudis) [inline backends/CLightningREST.ts:118]: turn this to `message: string`

File: backends/CLightningREST.ts [hunk 2]
```diff
@@ -115,6 +115,10 @@ export default class CLightningREST extends LND {
...
+        this.postRequest('/v1/utility/signMessage', {
+            message: `${message}`
```
Comment (kaloudis) [inline backends/CLightningREST.ts:120]: this can just be `message`

File: backends/Eclair.ts [hunk 3]
```diff
@@ -460,6 +460,11 @@ export default class Eclair {
             });
     };
 
+    signMessage = (message: any) =>
```
Comment (kaloudis) [inline backends/Eclair.ts:463]: `message: string`

File: backends/Eclair.ts [hunk 4]
```diff
@@ -460,6 +460,11 @@ export default class Eclair {
...
+        this.api('signmessage', {
+            msg: `${message}`
```
Comment (kaloudis) [inline backends/Eclair.ts:465]: `msg: message`

File: backends/LND.ts [hunk 5]
```diff
@@ -246,8 +246,10 @@ export default class LND {
                 urlParams[1]}`
         );
     getForwardingHistory = (data: any) => this.postRequest('/v1/switch', data);
-    signMessage = (data: any) =>
-        this.postRequest('/v2/signer/signmessage', data);
+    signMessage = (message: any) =>
```
Comment (kaloudis) [inline backends/LND.ts:249]: `message: string`

File: backends/LND.ts [hunk 6]
```diff
@@ -246,8 +246,10 @@ export default class LND {
...
+        this.postRequest('/v1/signmessage', {
+            msg: `${message}`
```
Comment (kaloudis) [inline backends/LND.ts:251]: `msg: message`

File: backends/CLightningREST.ts (hunk 1 above)
Comment (pseudozach) [reply backends/CLightningREST.ts:118]: Sorry I was just going with what was above in similar code. I'll change all of them to regular variable.

File: backends/CLightningREST.ts (hunk 2 above)
Comment (pseudozach) [reply backends/CLightningREST.ts:120]: I'll change all of them to regular variable.

File: backends/CLightningREST.ts [hunk 7]
```diff
@@ -115,6 +115,10 @@ export default class CLightningREST extends LND {
         });
     getRoutes = () => this.getRequest('N/A');
     getUTXOs = () => this.getRequest('/v1/listFunds');
+    signMessage = (message: string) =>
+        this.postRequest('/v1/utility/signMessage', {
+            message: message
```
Comment (hsjoberg) [inline backends/CLightningREST.ts:120]: Is `message` here base64 encoded?

According to the API docs, it should be encoded in base64 when using the REST API:
https://api.lightning.community/#v1-signmessage

Also small nit depedning on how you do code style in the project: message could use the shorthand syntax: `{ message }`.

**Review by hsjoberg — CHANGES_REQUESTED** [review]

File: backends/LND.ts [hunk 8]
```diff
@@ -246,8 +246,10 @@ export default class LND {
                 urlParams[1]}`
         );
     getForwardingHistory = (data: any) => this.postRequest('/v1/switch', data);
-    signMessage = (data: any) =>
-        this.postRequest('/v2/signer/signmessage', data);
+    signMessage = (message: string) =>
+        this.postRequest('/v1/signmessage', {
+            msg: message
```
Comment (hsjoberg) [inline backends/LND.ts:251]: Is message here base64 encoded?

According to the lnd's API docs, it should be encoded in base64 when using the REST API:
https://api.lightning.community/#v1-signmessage

**Review by hsjoberg — CHANGES_REQUESTED** [review]

File: backends/LND.ts (hunk 8 above)
Comment (pseudozach) [reply backends/LND.ts:251]: yes, the string is passed through the function below which encodes base64 and then sent to the API endpoints.
>Base64Utils.btoa(text)

**Comment by kaloudis** [comment]
Can we create a new function in the backends called `supportsMessageSigning`?

**Comment by pseudozach** [comment]
> Can we create a new function in the backends called `supportsMessageSigning`?

done please check

File: stores/MessageSignStore.ts [hunk 9]
```diff
@@ -17,6 +17,7 @@ export default class MessageSignStore {
     public signMessage = (text: string) => {
         this.loading = true;
         const body = Base64Utils.btoa(text);
+        console.log("text, body ", text, body);
```
Comment (kaloudis) [inline stores/MessageSignStore.ts]: console log

File: android/build.gradle [hunk 10]
```diff
@@ -3,7 +3,7 @@
 buildscript {
     ext {
         buildToolsVersion = "28.0.3"
-        minSdkVersion = 26
+        minSdkVersion = 25
```
Comment (kaloudis) [inline android/build.gradle]: why the sdkVersion change?

File: android/app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -1,8 +1,10 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:tools="http://schemas.android.com/tools"
     package="app.zeusln.zeus">
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.CAMERA" />
+    <uses-sdk tools:overrideLibrary="com.sifir.tor"/>
```
Comment (kaloudis) [inline android/app/src/main/AndroidManifest.xml]: are these changes required?

**Review by kaloudis — APPROVED** [review]

**Pull request merged**